Core/src/org/sleuthkit/autopsy/timeline/actions/AddManualEvent.java

```diff
@@ -0,0 +1,299 @@
+
+        setEventHandler(actionEvent -> {
+            //shoe the dialog and if it completed normally add the event.
+            new EventCreationDialog(controller, epochMillis).showAndWait().ifPresent(this::addEvent);
+        });
+    }
+
+    /**
+     * Use the supplied ManualEventInfo to make an TSK_TL_EVENT artifact which
+     * will trigger adding a TimelineEvent.
+     *
+     * @param eventInfo The ManualEventInfo with the info needed to create an
+     *                  event.
+     *
+     * @throws IllegalArgumentException
+     */
+    @NbBundle.Messages({
+        "AddManualEvent.createArtifactFailed=Failed to create artifact for event.",
+        "AddManualEvent.postArtifactFailed=Failed to post artifact to blackboard."
+    })
+    private void addEvent(ManualEventInfo eventInfo) throws IllegalArgumentException {
+        SleuthkitCase sleuthkitCase = controller.getEventsModel().getSleuthkitCase();
+
+        try {
+            //Use the current examiners name plus a fixed string as the source / module name.
+            String source = MANUAL_CREATION + ": " + sleuthkitCase.getCurrentExaminer().getLoginName();
+
+            BlackboardArtifact artifact = sleuthkitCase.newBlackboardArtifact(TSK_TL_EVENT, eventInfo.datasource.getId());
+            artifact.addAttributes(asList(
+                    new BlackboardAttribute(
+                            TSK_TL_EVENT_TYPE, source,
+                            EventType.USER_CREATED.getTypeID()),
+                    new BlackboardAttribute(
+                            TSK_DESCRIPTION, source,
+                            eventInfo.description),
+                    new BlackboardAttribute(
+                            TSK_DATETIME, source,
+                            eventInfo.time)
+            ));
+            try {
+                sleuthkitCase.getBlackboard().postArtifact(artifact, source);
+            } catch (Blackboard.BlackboardException ex) {
+                logger.log(Level.SEVERE, "Error posting artifact to the blackboard.", ex);
+                new Alert(Alert.AlertType.ERROR, Bundle.AddManualEvent_postArtifactFailed(), ButtonType.OK).showAndWait();
+            }
+        } catch (TskCoreException ex) {
+            logger.log(Level.SEVERE, "Error creatig new artifact.", ex);
+            new Alert(Alert.AlertType.ERROR, Bundle.AddManualEvent_createArtifactFailed(), ButtonType.OK).showAndWait();
+        }
+    }
+
+    /** The dialog that allows the user to enter the event information. */
+    private static class EventCreationDialog extends Dialog<ManualEventInfo> {
+
+        /** Custom DialogPane defined below. */
+        private final EventCreationDialogPane eventCreationDialogPane;
+
+        EventCreationDialog(TimeLineController controller, Long epochMillis) {
+            this.eventCreationDialogPane = new EventCreationDialogPane(controller, epochMillis);
+            setTitle("Add Event");
+            setDialogPane(eventCreationDialogPane);
+
```

Comment:
In reference to my above comment you should be able to put code here, before the dialog is shown.  JDialog has a function setLocationRelativeTo(), I know this is not a JDialog, but I would assume there is something similar.